faillint/faillint.go

```diff
@@ -201,3 +211,67 @@ func isTopName(n ast.Expr, name string) bool {
 	id, ok := n.(*ast.Ident)
 	return ok && id.Name == name && id.Obj == nil
 }
+
+func parseDirective(s string) (cmd string, args []string) {
+	if !strings.HasPrefix(s, "//lint:") {
+		return "", nil
+	}
+	s = strings.TrimPrefix(s, "//lint:")
+	fields := strings.Split(s, " ")
+	return fields[0], fields[1:]
+}
+
+func fileIsIgnored(f *ast.File) bool {
+	if f.Doc == nil {
+		return false
+	}
+	for _, comment := range f.Doc.List {
+		cmd, args := parseDirective(comment.Text)
+		if cmd == "file-ignore" && len(args) > 0 && args[0] == "faillint" {
+			return true
+		}
+	}
+	return false
+}
+
+func importIsIgnored(s *ast.ImportSpec) bool {
+	if s.Doc == nil {
+		return false
+	}
+	for _, comment := range s.Doc.List {
+		if commentHasIgnoreDirective(comment) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I think both `fileIsIgnored` and `usageIsIgnored` can be combined into a single funtion, such as: 

```go
func IsIgnored(cm *ast.CommentGroup, option string) bool {
 	if cm == nil {
 		return false
 	}
 	for _, comment := range cm.List {
 		cmd, args := parseDirective(comment.Text)
 		if cmd == option && len(args) > 0 && args[0] == "faillint" {
 			return true
 		}
 	}
 	return false
 }
```

and then use it like:


```go
isIgnored(file.Doc, "file-ignore)
```

or 

```go
isIgnored(spec.Doc, "ignore")
```

This also means we can get rid of `commentHasIgnoreDirective`

Reply:
As I switched to this style I felt like the function should not have ignore in the name because it could technically be more broad and because it stuttered.  I did shorten the names but because of this deviated from your suggestions.  Please, of course, let me know what you think of the names once you see it in its new form.